pkg/bigquery/sanitize_test.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bigquery
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestValidateGCPProjectID(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name    string
+		input   string
+		wantErr bool
+	}{
+		{
+			name:    "happy_path",
+			input:   "github4-metrics",
+			wantErr: false,
+		},
+		{
+			name:    "space_fails",
+			input:   " ",
+			wantErr: true,
+		},
+		{
+			name:    "end_with_hyphen_fails",
+			input:   "foobar-",
+			wantErr: true,
+		},
+		{
+			name:    "start_with_hyphen_fails",
+			input:   "-foobar",
+			wantErr: true,
+		},
+		{
+			name:    "start_with_number_fails",
+			input:   "4foobar",
+			wantErr: true,
+		},
+		{
+			name:    "too_short_fails",
+			input:   "short",
+			wantErr: true,
+		},
+		{
+			name:    "six_chars_succeeds",
+			input:   "abcdef",
+			wantErr: false,
+		},
+		{
+			name:    "end_with_number_succeeds",
+			input:   "foobar4",
+			wantErr: false,
+		},
+		{
+			name:    "empty_string_fails",
+			input:   "",
+			wantErr: true,
+		},
+		{
+			name:    "underscore_fails",
+			input:   "github_metrics",
+			wantErr: true,
+		},
+		{
+			name:    "closing_backtick_fails",
+			input:   "github_metrics`",
+			wantErr: true,
+		},
+		{
+			name:    "closing_quote_fails",
+			input:   "github_metrics\"",
+			wantErr: true,
+		},
+		{
+			name:    "semi_colon_fails",
+			input:   "github_metrics;",
+			wantErr: true,
+		},
+		{
+			name:    "space_fails",
+			input:   "github_metrics ",
+			wantErr: true,
+		},
+		{
+			name:    "dot_fails",
+			input:   "github.metrics",
+			wantErr: true,
+		},
+		{
+			name:    "unicode_fails",
+			input:   "様様様様様様",
+			wantErr: true,
+		},
+		{
+			name:    "30_succeeds",
+			input:   strings.Repeat("a", 30),
+			wantErr: false,
+		},
+		{
+			name:    "too_long_fails",
+			input:   strings.Repeat("a", 31),
+			wantErr: true,
+		},
+		{
+			name:    "multiline_string_fails",
+			input:   "foobar\nbarfoo",
+			wantErr: true,
+		},
+	}
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			if err := ValidateGCPProjectID(tc.input); (err != nil) != tc.wantErr {
+				t.Errorf("ValidateGCPProjectID(%q) error = %v, wantErr %v", tc.input, err, tc.wantErr)
+			}
+		})
```

Comment:
```suggestion
	}
	
	for _, tc := range cases {
		tc := tc
		
		t.Run(tc.name, func(t *testing.T) {
			t.Parallel()
			
			if err := ValidateGCPProjectID(tc.input); (err != nil) != tc.wantErr {
				t.Errorf("ValidateGCPProjectID(%q) error = %v, wantErr %v", tc.input, err, tc.wantErr)
			}
		})
```

(throughout all tests)